### DISCUSSION TITLE
Handling of work in progress pull requests

### DISCUSSION THREAD

**Discussion by jamesobutler** [Project-MONAI/MONAILabel#734] [General]
## **Is your feature request related to a problem? Please describe.**
I've observed in https://github.com/Project-MONAI/MONAILabel/pulls a varying method of indicating that a pull request still has some work to do or that it should not be merged. There is usage of "[WIP]" in the PR title which some GitHub apps can use by a status check, though there is no status check for this repo. There is a label called "PR_DO_NOT_MERGE" and also a label "WORK_IN_PROGRESS". In the [Contributing.md](https://github.com/Project-MONAI/MONAILabel/blob/main/CONTRIBUTING.md) there isn't a clear indication of a preferred method.
![image](https://user-images.githubusercontent.com/15837524/162336059-835b8212-36ea-4c77-a203-a0438d41160c.png)

## **Describe the solution you'd like**
Instead of the various methods, I would suggest the more robust solution of using GitHub's tool of having "Draft" PRs (https://github.blog/changelog/2020-04-08-convert-pull-request-to-draft/). This functionality can be used to indicate that there is still "Work in Progress" and that no review is being requested at this time (see image below). Also it can be used when there is an approving review to prevent merging. Labels or PR text by their own cannot prevent actual merging and rely on humans reading and following instructions. It appears that the MONAI toolkit has already adopted this method https://github.com/Project-MONAI/MONAI/pulls.

![image](https://user-images.githubusercontent.com/15837524/162336504-bcbc8789-ad0b-4ad1-add7-c60a6b46f339.png)

## **Describe alternatives you've considered**
Another method is to use https://github.com/apps/wip, which when it detects "WIP" in the PR title, will set a status check to "in progress" which will prevent merging (set this as a required check) even when there is an approving review. Admin's can of course override this check.
![image](https://user-images.githubusercontent.com/15837524/162337147-b064c6e1-a98f-471a-ba8e-9fdba64c3255.png)

**Comment by diazandr3s** [comment]
Hi @jamesobutler,

Thanks for opening this discussion. It's a good point. 
Will consider your suggestions for the next PRs :)

Thanks again!!
